lib/deaf_grandma.rb

```diff
@@ -19,10 +20,19 @@ def run!
 
 
   def speak(input)
-
-    #Implement your code here <<<<<<<<<
-
+     
+     if input != input.upcase
+          "SPEAK UP SONNY!"
+    elsif input == "BYE"
+            @bye_counter = @bye_counter += 1
```

Comment:
`@bye_counter += 1` by itself changes the value of `@bye_counter`, you don't need to assign the return value back into `@bye_counter`.